### DISCUSSION TITLE
SoilBase Clarification

### DISCUSSION THREAD

**Discussion by Gil4390** [useocl/use#67] [Q&A] [answered]
Hello Lars @h-man2 
Can you point out the files that are responsible for handling the input from the terminal \ from the GUI,
we want to find the place where "SoilBase.gpart" gets is used,  as far as we understand, SoilBase parses operation statements of a class, but we didn't find where the parse actually gets called.
Thank you.
Gil and Amiel.

**Comment by h-man2** [comment, marked as answer]
Hi Gil and Amiel,
the starting point for executing SOIL-statements from the USE-shell is the file `use-gui/src/main/java/org/tzi/use/main/shell/Shell.java`.
Have a look at the operation `cmdExec`.

https://github.com/useocl/use/blob/39991e51d8a89207182ee1275740b89b8333f3af/use-gui/src/main/java/org/tzi/use/main/shell/Shell.java#L596-L651

Best regards
Lars
